### PR TITLE
Add gemm add c sw test and ci

### PR DESCRIPTION
**Pull request by xiaoling-yi**
In this PR, we add the software test and ci for the latest gemm with a functionality of D = A * B + C.

### CODE REVIEW

**Review by JosseVanDelm — COMMENTED** [review]
I feel like overall this PR is okay, but why we are adding yet another system to the mix? We now have more than 3 different gemms in CI, it is becoming a bit messy and confusing in my opinion. Every time we add a new option, our code base grows with 
Furthermore, this seems to be copy pasting a lot of stuff again and again. Can we avoid this somehow?

There seems to be a mistake in your waiting routine

File: target/snitch_cluster/sw/snax/streamer-gemm-add-c/src/snax-streamer-gemm-add-c-lib.c [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    write_csr(986 + 6, subtractions);
+}
+
+// Set CSR to start GEMM
+void set_block_gemm_start() { write_csr(987 + 6, 1); }
+
+// Poll until Streamer and GEMM accelerator finish
+void wait_streamer_gemm() {
+    write_csr(987 + 6, 0);
+    write_csr(987 + 6, 0);
```
[the remark this answers is not in the document]
Comment (xiaoling-yi) [reply target/snitch_cluster/sw/snax/streamer-gemm-add-c/src/snax-streamer-gemm-add-c-lib.c:91]: Yes, this is a work-around for reading the performance counter correctly...We can remove it now.

File: target/snitch_cluster/sw/snax/streamer-gemm-add-c/src/snax-streamer-gemm-add-c-lib.c [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+// Set CSR to start STREAMER
+void set_streamer_start() { write_csr(987, 1); }
+
+// Set GEMM configuration CSR
+void set_block_gemm_csr(int tempLoop0, int tempLoop1, int tempLoop2,
+                        int subtractions) {
+    // set loop bounds, from innermost to outermost, aka from K to N to M
+    write_csr(983 + 6, tempLoop0);
+    write_csr(984 + 6, tempLoop1);
+    write_csr(985 + 6, tempLoop2);
```
[the remark this answers is not in the document]
Comment (xiaoling-yi) [reply target/snitch_cluster/sw/snax/streamer-gemm-add-c/src/snax-streamer-gemm-add-c-lib.c:79]: solved!

File: target/snitch_cluster/sw/apps/snax-streamer-gemm-add-c/src/snax-streamer-gemm-add-c.c [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+            if (snrt_global_core_idx() == 0) {
+                for (int i = 0; i < M * N * meshRow * meshCol; i++) {
+                    if (local_d[i] !=
+                        D_golden[i + m * N2 * M * N * meshRow * meshCol +
+                                 n * M * N * meshRow * meshCol]) {
+                        err++;
+                    }
+                }
+                printf("GEMM on A * B + C finished. error: %d\n", err);
+            }
```
[the remark this answers is not in the document]
Comment (xiaoling-yi) [reply target/snitch_cluster/sw/apps/snax-streamer-gemm-add-c/src/snax-streamer-gemm-add-c.c:115]: The local_d will be overridden with a new result matrix very block matrix multiplication. Here is only a case study of how to use the gemm. For measuring performance, you can have other tricks.

**Comment by xiaoling-yi** [comment]
> I feel like overall this PR is okay, but why we are adding yet another system to the mix? We now have more than 3 different gemms in CI, it is becoming a bit messy and confusing in my opinion. Every time we add a new option, our code base grows with Furthermore, this seems to be copy pasting a lot of stuff again and again. Can we avoid this somehow?
 
We need at least the streamer-gemm and this gemm-add-c version for now because of the publication purpose. The very old gemm can be removed if no one is using it.

> There seems to be a mistake in your waiting routine

The waiting is done by reading the start CSR. It will block if the accelerator is not ready

File: target/snitch_cluster/sw/apps/snax-streamer-gemm-add-c/src/snax-streamer-gemm-add-c.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (JosseVanDelm) [reply target/snitch_cluster/sw/apps/snax-streamer-gemm-add-c/src/snax-streamer-gemm-add-c.c:115]: I see. Makes sense to test it like this. Thanks!

File: util/sim/snax_utils.py [hunk 4]
```diff
@@ -183,6 +183,80 @@ def block_gemm_golden_model(
     return d
 
 
+def tiled_block_gemm_golden_model(
+    m2, k2, n2, m, k, n, row, size, col, a, b, subtraction_a, subtraction_b, c
```
[the remark this answers is not in the document]
Comment (JosseVanDelm) [reply util/sim/snax_utils.py:208]: Thanks!

**Review by JosseVanDelm — APPROVED** [review]
Thanks!

**Pull request merged**